src/marshmallow/schema.py

```diff
@@ -501,7 +502,9 @@ def _call_and_store(getter_func, data, *, field_name, error_store, index=None):
             return error.valid_data or missing
         return value
 
-    def _serialize(self, obj: _T | typing.Iterable[_T], *, many: bool = False):
+    def _serialize(
+        self, obj: typing.Any | typing.Iterable[dict], *, many: bool = False
```

Comment:
`obj: typing.Any | typing.Iterable[dict]`

should have been

`obj: typing.Any | typing.Iterable[typing.Any]`

which anyway can be simplified as

`obj: typing.Any`

from mypy perspective at least, for a human `| typing.Iterable[typing.Any]` indicates a clearer intent about the `many` case.

Reply:
I just force-pushed a new commit with just `obj: typing.Any`.